packages/vuetify/src/components/VTabs/VTabs.tsx

```diff
@@ -86,36 +105,63 @@ export const VTabs = genericComponent()({
 
     useRender(() => {
       const slideGroupProps = VSlideGroup.filterProps(props)
+      const hasWindow = !!(slots.window || props.items.length > 0)
 
       return (
-        <VSlideGroup
-          { ...slideGroupProps }
-          v-model={ model.value }
-          class={[
-            'v-tabs',
-            `v-tabs--${props.direction}`,
-            `v-tabs--align-tabs-${props.alignTabs}`,
-            {
-              'v-tabs--fixed-tabs': props.fixedTabs,
-              'v-tabs--grow': props.grow,
-              'v-tabs--stacked': props.stacked,
-            },
-            densityClasses.value,
-            backgroundColorClasses.value,
-            props.class,
-          ]}
-          style={[
-            { '--v-tabs-height': convertToUnit(props.height) },
-            backgroundColorStyles.value,
-            props.style,
-          ]}
-          role="tablist"
-          symbol={ VTabsSymbol }
-        >
-          { slots.default ? slots.default() : parsedItems.value.map(item => (
-            <VTab { ...item } key={ item.text } />
-          ))}
-        </VSlideGroup>
+        <>
+          <VSlideGroup
+            { ...slideGroupProps }
+            v-model={ model.value }
+            class={[
+              'v-tabs',
+              `v-tabs--${props.direction}`,
+              `v-tabs--align-tabs-${props.alignTabs}`,
+              {
+                'v-tabs--fixed-tabs': props.fixedTabs,
+                'v-tabs--grow': props.grow,
+                'v-tabs--stacked': props.stacked,
+              },
+              densityClasses.value,
+              backgroundColorClasses.value,
+              props.class,
+            ]}
+            style={[
+              { '--v-tabs-height': convertToUnit(props.height) },
+              backgroundColorStyles.value,
+              props.style,
+            ]}
+            role="tablist"
+            symbol={ VTabsSymbol }
+          >
+            { slots.default?.() ?? items.value.map(item => (
+              slots[`header-item.${item.value}`]?.() ?? slots['header-item']?.() ?? (
```

Comment:
- Should we call these slots `tab` instead?
- Need to be passed the current item.